### PR TITLE
Add DatasetType.

### PR DESCRIPTION
**Pull request by eujern**
#45 Adding a data type type to the datasets

### CODE REVIEW

File: kloppy/domain/models/common.py [hunk 1]
```diff
@@ -224,7 +224,17 @@ class Metadata:
     provider: Provider
 
 
+class DatasetType(Enum):
```
Comment (eujern) [inline kloppy/domain/models/common.py:227]: Same naming convention as `EventType`.  There is however `DatasetType = TypeVar("DatasetType")` defined in line 118 of `kloppy/helpers.py` as well as `kloppy/domain/services/transformers/__init__.py`.  In the latter case, it's class scoped.  Could be confusing. Necessary to rename?

Comment (koenvo) [reply kloppy/domain/models/common.py:227]: Good you kept the naming convention! 

The code in `TypeVar` is used for typehinting. Would be good indeed to rename it. Let's use `DatasetT` for `helpers.py` and the transformers

Comment (eujern) [reply kloppy/domain/models/common.py:227]: Renamed.

**Comment by koenvo** [comment]
Sorry I missed this one: please add an assert to the tests to make sure the value is set correctly

**Comment by eujern** [comment]
Updated tests.

**Comment by koenvo** [comment]
Great!

**Pull request merged**